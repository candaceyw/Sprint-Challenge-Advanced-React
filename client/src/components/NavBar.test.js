import React from 'react'
import NavBar from './NavBar'
import { fireEvent, render, cleanup, act } from '@testing-library/react'
import renderer from 'react-test-renderer';
// import axiosMock from 'axios'
import '@testing-library/jest-dom/extend-expect';


test('change values via the fireEvent.change method', () => {
    const handleChange = jest.fn()
    const { container } = render(<input type="checkbox" onChange={() => setDarkMode(prevMode => !prevMode)} />)
    const input = container.firstChild
    fireEvent.change(input, { target: { value: false } })
    expect(input.value).toBe('false')
  })

  test('renders toggle', () => {
    // Arrange
    const { getByTestId } = render(<NavBar />);

  // Act
  const toggle = getByTestId(/Toggle/i);

  // Assert
  expect(toggle).toBeInTheDocument();
  })


  test('is clickable', () => {
    // Arrange
    const spy = jest.fn();
    const { getByTestId } = render(<NavBar />);
  
    // Act
    const checkbox = getByTestId(/checkbox/i);
    checkbox.onclick = spy;
    fireEvent.click(checkbox);
  
    // Assert
    expect(spy).toHaveBeenCalledTimes(1);
  })



describe('NavBar', () => {
    test('snapshot renders', () => {
      const component = renderer.create(<NavBar />);
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });