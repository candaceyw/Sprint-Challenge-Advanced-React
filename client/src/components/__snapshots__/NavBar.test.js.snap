// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar snapshot renders 1`] = `
<div
  className="dark-mode"
>
  <nav>
    <div
      className="toggle-container"
    >
      <span
        style={
          Object {
            "color": "grey",
          }
        }
      >
        ☀︎
      </span>
      <span
        className="toggle"
        data-testid="toggle"
      >
        <input
          checked={true}
          className="checkbox"
          data-testid="checkbox"
          id="checkbox"
          onChange={[Function]}
          type="checkbox"
          value={true}
        />
        <label
          htmlFor="checkbox"
        />
      </span>
      <span
        style={
          Object {
            "color": "slateblue",
          }
        }
      >
        ☾
      </span>
    </div>
  </nav>
</div>
`;
